src/JackTrip.cpp

```diff
@@ -589,21 +589,29 @@ void JackTrip::completeConnection()
         mAudioInterface->appendProcessPluginFromNetwork(i);
     }
 
-    VuMeter* outputMeterPlugin = new VuMeter(mNumAudioChansOut);
-    mAudioInterface->appendProcessPluginFromNetwork(outputMeterPlugin);
-    connect(outputMeterPlugin, &VuMeter::onComputedVolumeMeasurements, this,
-            &JackTrip::receivedOutputVolumeMeasurements);
-    mVuMeterValuesOut.resize(mNumAudioChansOut);
+#ifndef NO_GUI
+    if (mJackTripMode == CLIENT || mJackTripMode == CLIENTTOPINGSERVER) {
+        VuMeter* outputMeterPlugin = new VuMeter(mNumAudioChansOut);
+        mAudioInterface->appendProcessPluginFromNetwork(outputMeterPlugin);
+        connect(outputMeterPlugin, &VuMeter::onComputedVolumeMeasurements, this,
+                &JackTrip::receivedOutputVolumeMeasurements);
+        mVuMeterValuesOut.resize(mNumAudioChansOut);
+    }
+#endif  // NO_GUI
 
     for (auto& i : mProcessPluginsToNetwork) {
         mAudioInterface->appendProcessPluginToNetwork(i);
     }
 
-    VuMeter* inputMeterPlugin = new VuMeter(mNumAudioChansIn);
-    mAudioInterface->appendProcessPluginToNetwork(inputMeterPlugin);
-    connect(inputMeterPlugin, &VuMeter::onComputedVolumeMeasurements, this,
-            &JackTrip::receivedInputVolumeMeasurements);
-    mVuMeterValuesIn.resize(mNumAudioChansIn);
+#ifndef NO_GUI
+    if (mJackTripMode == CLIENT || mJackTripMode == CLIENTTOPINGSERVER) {
+        VuMeter* inputMeterPlugin = new VuMeter(mNumAudioChansIn);
```

Comment:
Sorry to ask again, but why is this line specifically in JackTrip.cpp? It seems like something the GUI should do. _Or_ a flag should be set by the GUI and Effects.h can handle it.

If you look in qjacktrip.cpp, all the other effects are instantiated there. That's for Classic mode, obviously, but we could do something similar in virtualstudio.cpp and maybe later provide controls and a view for classic mode.

Reply:
Realizing today that I repeated what you said to me in the other comment. 🤦 

I will go back there